rcl_action/src/rcl_action/action_server.c

```diff
@@ -693,6 +693,9 @@ rcl_ret_t
 rcl_action_notify_goal_done(
   const rcl_action_server_t * action_server)
 {
+  if (!rcl_action_server_is_valid(action_server)) {
```

Comment:
I assume this is to fix one of the new tests added? It's probably worth mentioning in the commit message that this PR contains a bug fix as well.

Reply:
Separated this and the changes to goal_state_machine.c into their own commit.